### PR TITLE
Developer Guide for GUI Enhancements

### PR DESCRIPTION
**Pull request by tiuweehan**

### CODE REVIEW

**Review by le0tan — APPROVED** [review]
lgtm

**Review by jiayushe — APPROVED** [review]
LGTM

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -318,6 +318,87 @@ _Figure 3. Sequence Diagram for the Execution of `find` Command_
...
+
+==== Current Implementation
+
+The state of the GUI is stored in a `GuiState` object. The `GuiState` object contains a `TabManager` object, which manages tab information such as the tabs that are open and the tabs that are currently selected.
+
+The information is linked to the `UI` through listeners. When the attributes of the `TabManager` object is updated, it triggers callback functions in the UI controllers that update the state of the UI displayed.
+
+The following commands facilitate the management of tabs:
+
+* `AlgoBase#switchTab()` - Switch between tabs within a specified Tab pane.
```
Comment (joloong) [inline docs/DeveloperGuide.adoc:336]: Inconsistency. `Tab` and `tab` in line 336 and 338.

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -318,6 +318,87 @@ _Figure 3. Sequence Diagram for the Execution of `find` Command_
...
+These operations are exposed in the `TabManager` class respectively as:
+
+* `SwitchTab()`: `TabManager#setDisplayTabIndex` and `TabManager#setDetailsTabIndex`
+* `OpenTab()`: `TabManager#addTab`
+* `CloseTab()`: `TabManager#removeTab`
+
+The following Activity Diagrams illustrate what happens when the user executes a `SwitchTabCommand` or `OpenTabCommand`
+
+image::gui/SwitchTabCommandActivityDiagram.png[]
+_Figure 1.1 Activity Diagram for the Execution of `switchtab` Command_
```
Comment (joloong) [inline docs/DeveloperGuide.adoc:349]: Not following camelCase for `switchtab`.

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -318,6 +318,87 @@ _Figure 3. Sequence Diagram for the Execution of `find` Command_
...
+
+image::gui/OpenTabCommandActivityDiagram.png[]
+_Figure 1.2 Activity Diagram for the Execution of `opentab` Command_
```
Comment (joloong) [inline docs/DeveloperGuide.adoc:352]: Figure 1.2 is almost the same as Figure 1.1 and there is very little value-add.

File: docs/DeveloperGuide.adoc [hunk 4]
```diff
@@ -318,6 +318,87 @@ _Figure 3. Sequence Diagram for the Execution of `find` Command_
...
+
+
+image::gui/SwitchTabsSequenceDiagram2.png[]
+_Figure 3.3 Sequence Diagram for reflecting the tab changes_
+
+Step 6: A listener was added to the `displayTabIndex` field when the application was initialized. When a change in the value is detected, it triggers the `selectTab(1)` method with the value of the new index passed as an argument. This updates the selected tab in the UI.
+
+==== Design Considerations
+
+Aspect: Implementation of Tab Logic
```
Comment (joloong) [inline docs/DeveloperGuide.adoc:381]: Could be 3.4.2.1. or stylize it.

**Review by joloong — COMMENTED** [review]
Well done. 😇

**Pull request merged**